simulator/main.rs

```diff
@@ -152,74 +538,94 @@ fn main() {
     env.io.print_stats();
 }
 
-fn process_connection(env: &mut SimulatorEnv, conn: &mut Rc<Connection>) -> Result<()> {
-    let management = env.rng.gen_ratio(1, 100);
-    if management {
-        // for now create table only
-        maybe_add_table(env, conn)?;
-    } else if env.tables.is_empty() {
-        maybe_add_table(env, conn)?;
-    } else {
-        let roll = env.rng.gen_range(0..100);
-        if roll < env.opts.read_percent {
-            // read
-            do_select(env, conn)?;
-        } else if roll < env.opts.read_percent + env.opts.write_percent {
-            // write
-            do_write(env, conn)?;
+fn property_insert_select(env: &mut SimulatorEnv, conn: &mut Rc<Connection>) {
+    // Get a random table
+    let table = env.rng.gen_range(0..env.tables.len());
+
+    // let table = &env.tables[table];
+
+    // Pick a random column
+    let column_index = env.rng.gen_range(0..env.tables[table].columns.len());
+    let column = &env.tables[table].columns[column_index].clone();
+
+    let mut rng = env.rng.clone();
+
+    // Generate a random value of the column type
+    let value = Value::arbitrary_from(&mut rng, &column.column_type);
+
+    // Create a whole new row
+    let mut row = Vec::new();
+    for (i, column) in env.tables[table].columns.iter().enumerate() {
+        if i == column_index {
+            row.push(value.clone());
         } else {
-            // delete
-            // TODO
+            let value = Value::arbitrary_from(&mut rng, &column.column_type);
+            row.push(value);
         }
     }
-    Ok(())
-}
 
-fn do_select(env: &mut SimulatorEnv, conn: &mut Rc<Connection>) -> Result<()> {
-    let table = env.rng.gen_range(0..env.tables.len());
-    let table_name = {
-        let table = &env.tables[table];
-        table.name.clone()
+    // Insert the row
+    let query = Query::Insert {
+        table: env.tables[table].name.clone(),
+        values: row.clone(),
+    };
+    let _ = get_all_rows(env, conn, query.to_string().as_str()).unwrap();
+    // Shadow operation on the table
+    env.tables[table].rows.push(row.clone());
```

Comment:
What if the row is not inserted in order? The idea was to keep the rows in the same order as the table would to improve lookup logic.

Reply:
Should the push be randomized?

Reply:
No, I just mean the env.tables[table].rows should probably be ordered the same way the rows would be ordered in disk. This way if we want to compare the rows we can do it quickly.